feat(integration)/youkassa payments list by Семиренко

AndVicSem112 opened the pull request:
## Описание изменений

Добавлен файл фикстуры для тестов `payments_list.json`, содержащий пример ответа API YooKassa для списка платежей. Эта фикстура используется в модульных тестах и демонстрационном скрипте для проверки интеграции `youkassa_get_payments` без реальных credentials.

## Тип изменений

- [ ] Исправление бага
- [x ] Новая функция
- [ ] Изменение документации
- [ ] Рефакторинг кода
- [x] Тесты
- [ ] Другое (опишите): фикстуры для тестов

## Чеклист

- [x] Код соответствует стилю проекта (JSON fixtures)
- [x] Я проверил свой код
- [x] Я добавил комментарии к коду, особенно в сложных местах (не применимо для fixtures)
- [ ] Я обновил документацию (если необходимо)
- [x] Мои изменения не генерируют новых предупреждений
- [x] Я добавил тесты, которые подтверждают мои исправления (тесты используют эту фикстуру)
- [x] Новые и существующие тесты проходят локально (см. инструкции ниже)

## Как протестировать

1. Переключиться на ветку `feat/youkassa-payments-list` (этот PR):
   ```bash
   git checkout feat/youkassa-payments-list
   ```
2. Убедиться, что фикстура присутствует: `backend/app/tests/fixtures/youkassa/payments_list.json`.
3. Запустить тесты интеграции (в основной ветке `feat/youkassa-tests-and-scripts`):
   ```bash
   python -m pytest backend/app/tests/integrations/test_yookassa_get_payments.py -q
   ```
4. Или запустить демонстрационный скрипт (в корне репозитория):
   ```bash
   python scripts/test_yookassa_connection.py
   ```

## Дополнительная информация

Фикстура покрывает типичный ответ `Payment.list` и используется вместе с другими фикстурами (payment_detail, receipt_create) в соответствующих ветках. Этот PR добавляет только данные фикстуры, код тестов и интеграции находятся в отдельном PR `feat/youkassa-tests-and-scripts`.

AndVicSem112 committed:
feat(integration): add youkassa get_payment get_payments create_receipt integrations and tests

AndVicSem112 committed:
Add YooKassa payments_list fixture

HadeSxGod1337 commented:
1. Интеграция использует youkassa провайдер, но такого нет. Т.к. интеграция использует api_key, то провайдер надо выбрать other + api_key стратегию.
2. Почему не стали создавать аккаунт тестовый на youkassa, чтобы проверить интеграции? Там не сложно.
3. Не показано как интеграция выглядит на фронтенде. По файлам вижу, что везде зарегистрирована, но все равно нужно видеть.
Прошу исправить.

AndVicSem112 committed:
docs: add live-run update — payments list returned empty

Added live-run note: GET /payments?limit=20 returned an empty list; recommend re-running with larger limit/date range; attached excerpt and referenced scripts/youkassa_live_results.json.

AndVicSem112 commented:
Создал личный кабинет в youkassa, прогнал тесты с ключом и Store ID. Закинул обновления. 
попробую создать бота на фронте, но мощность моего учебного ноутбука оставляет желать лучшего, контейнеры и те еле запускаются ) 

# Live-run comment — feat/youkassa-payments-list

**Live-run — youkassa_get_payments (GET /payments)**

Я выполнил live-прогон списка платежей с тестовыми креденшелами.

**Запрос (пример):**

```
GET https://api.yookassa.ru/v3/payments?limit=20
Authorization: Basic <base64(shop_id:api_key)>
```

**Live-ответ (фрагмент):**

```json
{
  "type": "list",
  "items": []
}
```

**Что получилось:** API вернул пустой список платежей (`items: []`), то есть в аккаунте нет платежей, подходящих под этот запрос.

**Почему:** платежи либо отсутствуют за указанный период / с текущими фильтрами, либо нужно увеличить выборку/фильтры.

**Рекомендации:**

- Повторить `GET /payments` с `limit=100` и/или добавить фильтры: `status` = `succeeded|paid|waiting_for_capture` и/или date range (last 30 days).
- Если есть конкретные `payment_id`, пришлите их — я выполню `GET /payments/{id}` и добавлю ответ в PR.

**Файлы / доказательства:**

- `.github/pr_descriptions/youkassa-live-results.md` (подробный отчёт)
- `scripts/youkassa_live_results.json` (экспорт ответа)

---
